torch/_torch_docs.py

```diff
@@ -8332,6 +8332,11 @@ def merge_dicts(*dicts):
 
 Reverse the order of a n-D tensor along given axis in dims.
 
+.. note::
+    Unlike `np.flip`, `torch.flip` returns a new Tensor.
```

Comment:
Let's tweak and simplify this just a bit.

"\`torch.flip\` makes a copy of :attr:\`input\`'s data. This is different from \`np.flip\`, which returns a view."

There are a couple reasons I want to suggest this change:

- it starts with torch.flip, no knowledge of NumPy required
- the phrase "a new tensor" may be ambiguous; a view appears like "a new tensor," too. We want to be explicit that torch.flip copies the data.
- it removes the time complexity discussion. That discussion is correct and relevant, but a little too detailed since it's not the only implication of this behavior. For example, the write semantics also change, but the level of detail of discussing the time complexity suggests that a change like that would be mentioned, too. Best simplify and avoid it.

Then this can be a template to substitute flip{ud, lr} into. Does that sound good to you, @kshitij12345?

Reply:
The first two points make sense.

I think it would be good to mention the time-complexity as it is a major gotcha https://github.com/pytorch/pytorch/issues/16424#issuecomment-580695672

Reply:
OK that's a good point.

Big O notation seems like a lot of machinery to bring to this documentation, though. Could we sneak in the performance discussion with something like (I'm also going to add an explicit reference to NumPy since "np" in the above is not defined):

"This is different from NumPy's `np.flip`, which quickly returns a view."

If that's too subtle, what about avoiding big O by doing:

"This is different from NumPy's `np.flip`, which returns a view. Since copying a tensor's data is more work than viewing that data, `torch.flip` is expected to be slower than `np.flip`."

Reply:
I believe `flip` would be used heavily on CV side in augmentation pipelines, so it might be better to mention about time-complexity.

Reply:
> Big O notation seems like a lot of machinery to bring to this documentation, though.

Agreed.

I would prefer,

"This is different from NumPy's np.flip, which returns a view. Since copying a tensor's data is more work than viewing that data, torch.flip is expected to be slower than np.flip."
as 
> which quickly returns a view.

Seems a bit ambiguous as to what is quick and what isn't

So change would be
```
.. note:: This is different from NumPy's np.flip, which returns a view. Since copying a tensor's data is more work than viewing that data, torch.flip is expected to be slower than np.flip.
   
```


Just a side note, numpy.flip documentation does mention that it is a constant-time operation. (Maybe we can say torch.flip is linear-time w.r.t number of elements in tensor)

https://numpy.org/doc/stable/reference/generated/numpy.flip.html

Reply:
".. note:: `torch.flip` makes a copy of :attr:`input`'s data. This is different from NumPy's np.flip, which returns a view in constant time. Since copying a tensor's data is more work than viewing that data, torch.flip is expected to be slower than np.flip."

Merging your suggestion about constant time + adding back the first sentence. How does this look?

Reply:
Yup. This feels better. Thanks!